src/components/Dialog.js

```diff
@@ -0,0 +1,139 @@
+import classnames from 'classnames';
+import { useEffect, useLayoutEffect, useRef, useState } from 'preact/hooks';
+
+import { IconButton, LabeledButton } from '../';
+import { SvgIcon } from './SvgIcon';
+
+let idCounter = 0;
+
+/**
+ * Return an element ID beginning with `prefix` that is unique per component instance.
+ *
+ * This avoids different instances of a component re-using the same ID.
+ *
+ * @param {string} prefix
+ */
+function useUniqueId(prefix) {
+  const [id] = useState(() => {
+    ++idCounter;
+    return `${prefix}-${idCounter}`;
+  });
+  return id;
+}
+
+/**
+ * @typedef {import("preact").ComponentChildren} Children
```

Comment:
@typedef {import('preact').ComponentChildren} Children

Reply:
This gives me `Generic type 'Ref' requires 1 type argument(s).ts(2314)`

Reply:
To confirm: single quotes here? Wish we could lint this!